api/src/main/java/org/pmiops/workbench/institution/InstitutionMapper.java

```diff
@@ -23,42 +14,17 @@
   @Mapping(target = "institutionId", ignore = true)
   DbInstitution modelToDb(Institution modelObject);
 
-  // userInstructions will be populated by setUserInstruction afterMapping
+  // these fields will be populated by @AfterMapping populateFromAuxTables
+  @Mapping(target = "emailDomains", ignore = true)
+  @Mapping(target = "emailAddresses", ignore = true)
   @Mapping(target = "userInstructions", ignore = true)
   Institution dbToModel(DbInstitution dbObject, @Context InstitutionService institutionService);
 
-  default List<String> toModelDomains(Set<DbInstitutionEmailDomain> dbDomains) {
-    return Optional.ofNullable(dbDomains).orElse(Collections.emptySet()).stream()
-        .map(DbInstitutionEmailDomain::getEmailDomain)
-        .sorted()
-        .collect(Collectors.toList());
-  }
-
-  default List<String> toModelAddresses(Set<DbInstitutionEmailAddress> dbAddresses) {
-    return Optional.ofNullable(dbAddresses).orElse(Collections.emptySet()).stream()
-        .map(DbInstitutionEmailAddress::getEmailAddress)
-        .sorted()
-        .collect(Collectors.toList());
-  }
-
-  // Swagger-generated Lists are null by default, so we should handle that
-  default Set<DbInstitutionEmailDomain> toDbDomainsWithoutInstitution(
-      @Nullable Collection<String> modelDomains) {
-    return Optional.ofNullable(modelDomains).orElse(Collections.emptySet()).stream()
-        .map(domain -> new DbInstitutionEmailDomain().setEmailDomain(domain))
-        .collect(Collectors.toSet());
-  }
-
-  default Set<DbInstitutionEmailAddress> toDbAddressesWithoutInstitution(
-      @Nullable Collection<String> modelAddresses) {
-    return Optional.ofNullable(modelAddresses).orElse(Collections.emptySet()).stream()
-        .map(address -> new DbInstitutionEmailAddress().setEmailAddress(address))
-        .collect(Collectors.toSet());
-  }
-
   @AfterMapping
-  default void setUserInstruction(
+  default void populateFromAuxTables(
       @MappingTarget Institution target, @Context InstitutionService institutionService) {
+    target.setEmailDomains(institutionService.getEmailDomains(target.getShortName()));
```

Comment:
I believe you can do this in your main mapping with attributes like `@Mapping(source = "institutionService.emailDomains", target = "emailDomains"`, unless I've missed something.

Reply:
How would that work?  You'd have to specify the source parameter to institutionService.getEmailDomains() as well.  I have not seen examples like this.

Reply:
Oh, I see.

By the way, I _think_ you can put `InstitutionService.class` in your `uses` list so you don't need to use `@Context`.